### PR TITLE
VR-9655: Enable code versioning with manual values, no autocapture

### PR DESCRIPTION
**Pull request by convoliution**
`Git(autocapture=True)` (default, current behavior) will check the local git env for info about the current commit.  
`Git(autocapture=False)` will not check for anything, instead directly passing user-input values as the git version.
`log_code(autocapture=True)` in git mode is the same as `Git(autocapture=True)`.  
`log_code(autocapture=False)` in git mode is the same as `Git(autocapture=False)`.  
`log_code(autocapture)` isn't applicable to non-git mode, because the client needs to be able to find a notebook file.

---
For `Git()`: change `_autocapture` to `autocapture`, and add logic to allow user to selectively pass in arguments.
For `log_code()`: add `autocapture`

### CODE REVIEW

File: client/verta/verta/code/_git.py [hunk 1]
```diff
@@ -45,25 +45,38 @@ class Git(_code._Code):
         )
 
     """
-    def __init__(self, repo_url=None, branch=None, tag=None, commit_hash=None, _autocapture=True):
```
Comment (convoliution) [inline client/verta/verta/code/_git.py:48]: remember to enable **Hide whitespace changes**

Comment (convoliution) [inline client/verta/verta/code/_git.py:48]: remember to enable **Hide whitespace changes**

File: client/verta/verta/_tracking/entity.py [hunk 2]
```diff
@@ -164,7 +164,7 @@ def _create_proto(cls, conn, *args, **kwargs):
     def _create_proto_internal(cls, conn, ctx, name, desc=None, tags=None, attrs=None, date_created=None, **kwargs):  # recommended params
         raise NotImplementedError
 
-    def log_code(self, exec_path=None, repo_url=None, commit_hash=None, overwrite=False):
+    def log_code(self, exec_path=None, repo_url=None, commit_hash=None, overwrite=False, autocapture=True):
```
Comment (convoliution) [inline client/verta/verta/_tracking/entity.py:167]: remember to enable **Hide whitespace changes**

File: client/verta/verta/_tracking/entity.py [hunk 3]
```diff
@@ -382,8 +382,8 @@ def get_code(self):
             Either:
                 - a dictionary containing Git snapshot information with at most the following items:
                     - **filepaths** (*list of str*)
-                    - **repo** (*str*) – Remote repository URL
-                    - **hash** (*str*) – Commit hash
+                    - **repo_url** (*str*) – Remote repository URL
+                    - **commit_hash** (*str*) – Commit hash
```
Comment (convoliution) [inline client/verta/verta/_tracking/entity.py:386]: Slightly unrelated, but I noticed this docstring was incorrect.

**Review by conradoverta — APPROVED** [review]
LGTM

**Pull request merged**